clientv3/client.go

```diff
@@ -280,9 +291,60 @@ func newClient(cfg *Config) (*Client, error) {
 		logger.Set(log.New(ioutil.Discard, "", 0))
 	}
 
+	if cfg.AutoSync > time.Duration(0) {
+		go client.autoSync()
+	}
 	return client, nil
 }
 
+func (c *Client) autoSync() {
+	if c.cfg.AutoSync == time.Duration(0) {
+		return
+	}
+	for {
+		select {
+		case <-time.After(c.cfg.AutoSync):
+			c.sync()
+		case <-c.ctx.Done():
+			return
+		}
+	}
+}
+
+func (c *Client) sync() {
+	mresp, err := c.MemberList(c.ctx)
+	if err != nil {
+		logger.Println(err)
+		return
+	}
+	epm := make(map[string]struct{})
+	for _, m := range mresp.Members {
+		for _, ep := range m.ClientURLs {
+			epm[ep] = struct{}{}
+		}
+	}
+	var eps []string
+	for ep := range epm {
+		eps = append(eps, ep)
+	}
+
+	c.balancer.mu.Lock()
+	defer c.balancer.mu.Unlock()
+	if len(c.balancer.lastAllEps) == 0 { // first sync
+		c.balancer.lastAllEps = epm
+	}
+	if !reflect.DeepEqual(c.balancer.lastAllEps, epm) { // membership change
+		c.balancer.upEps = epm
+		addrs := make([]grpc.Address, len(epm))
+		for i, ep := range eps {
+			addrs[i].Addr = getHost(ep)
+		}
+		c.balancer.addrs = addrs
+		c.balancer.pinAddr = eps[0]
```

Comment:
Does this work? I believe grpc needs a message posted to the notify channel before it'll establish connections on new addresses.